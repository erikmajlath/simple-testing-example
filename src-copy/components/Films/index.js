import React from 'react'
import styled from 'styled-components'

const API_ENDPOINT = 'http://swapi.co/api'

/* eslint-disable prettier/prettier */
export const FancyItem = styled.li`
  background-color: pink;
`
/* eslint-enable prettier/prettier */

export default class Films extends React.Component {
  state = {
    films: [],
  }

  // Do not do this!
  componentDidMount() {
    return fetch(`${API_ENDPOINT}/films`)
      .then(response => response.json())
      .then(films =>
        this.setState({
          films: films.results.map(film => film.title),
        })
      )
      .catch(err => console.log(err))
  }

  render() {
    return (
      <ul>
        {this.state.films.map(film =>
          <FancyItem key={film}>
            {film}
          </FancyItem>
        )}
      </ul>
    )
  }
}
