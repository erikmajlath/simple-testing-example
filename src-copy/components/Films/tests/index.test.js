import React from 'react'
import { mount, shallow } from 'enzyme'
import Films, { FancyItem } from '../index'
// import 'jest-styled-components'

const getResponse = results => new window.Response(JSON.stringify({ results }))
const results = [{ title: 'one' }, { title: 'two' }]

fetch = jest.fn()

describe('Films', () => {
  let wrapper

  it('should match snapshot', () => {
    fetch.mockImplementation(() => Promise.resolve(getResponse(results)))
    wrapper = mount(<Films />)
    return wrapper.instance().componentDidMount().then(() => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  it('should load films', () => {
    wrapper = shallow(<Films />)
    fetch.mockImplementation(() => Promise.resolve(getResponse(results)))
    return wrapper.instance().componentDidMount().then(() => {
      expect(wrapper.find({ children: 'one' }).length).toBe(1)
      expect(wrapper.find({ children: 'two' }).length).toBe(1)
      // fetch.mockReset()
    })
  })

  it('should call fetch on mount', () => {
    fetch.mockImplementationOnce(() => Promise.resolve(getResponse(results)))
    wrapper = mount(<Films />)

    expect(fetch).toBeCalledWith('http://swapi.co/api/films')
    fetch.mockReset()
  })
})

// jest.mock('utils/request', () =>
//   jest.fn().mockImplementation(url => {
//     const name = url === 'posts/1' ? 'firstPublisher' : 'anotherPublisher'
//     return Promise.resolve({
//       source: { publisher: { name } },
//       media: [{ original_url: { url: 'imageUrl' } }],
//       link: 'postLink',
//     })
//   })
// )

// jest.mock('containers/ModalManager', () => ({
//   popup: jest.fn(),
// }))
