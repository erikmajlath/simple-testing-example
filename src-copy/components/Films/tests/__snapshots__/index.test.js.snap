// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films should match snapshot 1`] = `
<Films>
  <ul>
    <styled.li>
      <li
        className="sc-bdVaJa bPamRI"
      >
        one
      </li>
    </styled.li>
    <styled.li>
      <li
        className="sc-bdVaJa bPamRI"
      >
        two
      </li>
    </styled.li>
  </ul>
</Films>
`;
